robocop/checkers/errors.py

```diff
@@ -217,6 +217,15 @@
         severity=RuleSeverity.ERROR,
         version=">=5.0",
     ),
+    "0415": Rule(
+        rule_id="0415",
+        name="invalid-section-in-resource",
+        msg="Resource file with '{{ section_name }}' section is invalid",
+        docs="""
+        Resource file cannot contain ``Test Cases`` or ``Tasks`` sections.
```

Comment:
For the docs we can then use a bit more extended sentence like:
"The higher-level structure of resource files is the same as that of test case files, but they can't contain ``Test Cases``  or ``Tasks`` sections."